src/com/company/Zadania.java

```diff
@@ -1332,5 +1332,29 @@ public void zadanie43() {
             System.out.print(x + ",");
         }
     }
+
+    //Write a program, which will find all such numbers between 1000 and 3000 (both included)
+    // such that each digit of the number is an even number.
+    //The numbers obtained should be printed in a comma-separated sequence on a single line.
+
+    public void zadanie44() {
+        int helper;
+        int digit;
+        int counter = 0;
+        for (int i = 1000; i < 3001; i++) {
+            helper = i;
+            for (int j = 0; j < 4; j++) {
+                digit = helper % 10;
+                if (digit % 2 == 0) {
+                    counter++;
+                }
+                helper = helper / 10;
+            }
+            if (counter == 4) {
+                System.out.print(i + " ,");
+                counter = 0;
+            } else counter = 0;
```

Comment:
counter powinien sie zawsze zerowac na poczatku tam gdize helper = i
w ogole bardziej czytelnie byloby 
counter = 4 ,
if (counter ==0 ) print)

Reply:
a jeszcze lepiej 

private boolean isEveryDigitEven(int number) {
  tutaj logika sprawdzania:
 return true/false;
}

zadanie44() {
  for (int i = 3000; ...) {
    if (isEveryDigitEven(i)) {
       print(i)
} 

..
a to czy coutner = 4 to mozna String.valueOf(liczba).length